### DISCUSSION TITLE
New stable release, or docs to install from source?

### DISCUSSION THREAD

**Discussion by Scramblejams** [fastlane/fastlane#19610] [Q&A] [answered]
(NB: Ruby noob here. Apologies in advance if I've missed the docs I'm asking about.)

Hi! Thanks for your work on fastlane! It looks to a very useful piece of software. I'm attempting to use it to deploy an iOS app.

However, I was blocked by [this issue](https://github.com/fastlane/fastlane/issues/19572). Was very happy to see it resolved today. So I'd like to try the fix. However, "brew install --HEAD fastlane" fails, I guess because head builds are not uploaded to rubygems.org. I had then hoped for installation docs that covered installing from source, but I [don't see those](https://docs.fastlane.tools/getting-started/ios/setup/), so I'm not-sufficiently-successfully bumbling my way through a Gemfile definition.

Could we get either a stable release that includes this fix, or an added page in the docs that covers building the gem from source?

Thanks!

**Comment by lucgrabowski** [comment, marked as answer]
Hi, fastlane 2.198 was released several hours ago, it is available on rubygems.org.

You can update it by running `bundle update fastlane` if you use `Gemfile`
 (please remember to commit `Gemfile.lock` to source code repository).

If you want to install gem from source code from `fastlane` repository or a fork, you can read how to do it in `bundler`'s docs: https://bundler.io/guides/git.html

Just to give you an example, to install code from a certain branch you can update `Gemfile` with a similar definition:
```ruby
gem "fastlane", git: "https://github.com/<fastlane repository or fork>/fastlane.git", branch: "<BRANCH_NAME>"
```

**Reply by Scramblejams** [reply]
The new stable release appears to have gotten me past that point, thank you! For posterity, my Gemfile had to look like this:

```
gem 'fastlane', git: 'https://github.com/fastlane/fastlane', branch: 'master', ref: '1879cc213d247ad0f96cc0a5adb3ab4632e1b186'

source 'https://rubygems.org' do
    gem 'xcpretty'
end
```

...or it would complain about xcpretty and fall over. Then the installation appeared to work correctly, but running it yielded the following error:

```
Traceback (most recent call last):
        2: from ./.bundle/ruby/2.6.0/fastlane-1879cc213d24/bin/fastlane:12:in `<main>'
        1: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- fastlane/cli_tools_distributor (LoadError)
```

I'm told that's an indicator that my dependencies are being mixed between system Ruby and the brew installation of fastlane I had put in earlier.

I'll close this now since the new stable release is out and contains the fix I needed to get unblocked. Thanks.
